src/cmd/watch.rs

```diff
@@ -179,3 +187,50 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ignore::gitignore::GitignoreBuilder;
+    use std::env;
+
+    fn path_to_buf(root: &str, file_name: &str) -> PathBuf {
+        let mut path = PathBuf::from(root);
+        path.push(file_name);
+        path
+    }
+
+    #[test]
+    fn test_filter_ignored_files() {
+        let current_dir = env::current_dir().unwrap();
+
+        let ignore = GitignoreBuilder::new(&current_dir)
+            .add_line(None, "*.html")
+            .unwrap()
+            .build()
+            .unwrap();
+        let should_remain = path_to_buf(current_dir.to_str().unwrap(), "record.text");
+        let should_filter = path_to_buf(current_dir.to_str().unwrap(), "index.html");
```

Comment:
I'm not entirely clear why the `path_to_buf` helper exists, as this should all be doable with the regular path APIs. This is taking a Path, converting it to a string, then converting it back to a PathBuf. Can you rework these to avoid that helper?

```suggestion
        let should_remain = current_dir.join("record.text");
        let should_filter = current_dir.join("index.html");
```

Reply:
Fix 40c034e. Thanks for feedback, this is much way cleaner.